background.ts

```diff
@@ -576,46 +576,59 @@ async function getVersion() {
 }
 
 /**
- * assume sync activities aren't going to be costly for a UI app.
+ * Manages the state of integration symlinks.
  * @param name -- basename of the resource to link
- * @param state -- true to symlink, false to delete
+ * @param desiredPresent -- true to symlink, false to delete
  */
-async function linkResource(name: string, state: boolean): Promise<Error | null> {
+async function linkResource(name: string, desiredPresent: boolean): Promise<void> {
   const linkPath = path.join(paths.integration, name);
 
-  let err: Error | null = await new Promise((resolve) => {
-    fs.mkdir(paths.integration, { recursive: true }, resolve);
-  });
-
-  if (err) {
-    console.error(`Error creating the directory ${ paths.integration }: ${ err.message }`);
-
-    return err;
+  try {
+    await fs.promises.mkdir(paths.integration, { recursive: true });
+  } catch (error: any) {
+    console.error(`Error creating integrations directory ${ paths.integration }: ${ error.message }`);
   }
 
-  if (state) {
-    err = await new Promise((resolve) => {
-      fs.symlink(resources.executable(name), linkPath, 'file', resolve);
-    });
-
-    if (err) {
-      console.error(`Error creating symlink for ${ linkPath }: ${ err.message }`);
-
-      return err;
+  if (desiredPresent) {
+    try {
+      await fs.promises.symlink(resources.executable(name), linkPath);
+    } catch (error: any) {
+      console.warn(`Failed to create symlink ${ linkPath }: ${ error.message }`);
     }
-  } else {
-    err = await new Promise((resolve) => {
-      fs.unlink(linkPath, resolve);
-    });
+  } else if (await isManagedIntegration(linkPath)) {
+    try {
+      await fs.promises.unlink(linkPath);
+    } catch (error: any) {
+      if (error.code !== 'ENOENT') {
+        console.error(`Error unlinking symlink ${ linkPath }: ${ error.message }`);
+      }
+    }
+  }
+}
 
-    if (err) {
-      console.error(`Error unlinking symlink for ${ linkPath }: ${ err.message }`);
+/**
+ * Tests whether a path is an integration symlink that is safe to delete.
+ * Can only return true when running RD as an AppImage.
+ * @param pathToCheck -- absolute path to the filesystem node that we want to check
+ * for being an integration symlink
+ */
+async function isManagedIntegration(pathToCheck: string): Promise<boolean> {
+  if (!process.env['APPIMAGE']) {
+    return false;
+  }
 
-      return err;
+  let linkedTo: string;
+
+  try {
+    linkedTo = await fs.promises.readlink(pathToCheck);
+  } catch (error: any) {
+    if (error.code === 'EINVAL') {
```

Comment:
Would we need to guard against `ENOENT` here too? (`EINVAL` would be a file exists, but is not a symlink.)

Reply:
Shouldn't **any** error result in a `false` return? I would think `linkedTo` is undefined when an error has ben thrown, so the following comparison should return `false` anyways.

Reply:
Sorry, I was confused, but the point still remains: why would you want to throw an error? This is being called when the app shuts down, so what value would the user get from getting an error about reading a symlink?

Unless the `pathToCheck` is a symlink that points to the right location, this function should just return `false` and not throw an error.

Reply:
Yeah that's a good idea, done.

Reply:
Sorry Jan, for some reason github didn't display your comments when I looked at it yesterday. I will make that change.